var/spack/repos/builtin/packages/py-dask/package.py

```diff
@@ -27,11 +28,11 @@ class PyDask(PythonPackage):
     variant('distributed', default=True, description='Install requirements for dask.distributed')
     variant('diagnostics', default=False, description='Install requirements for dask.diagnostics')
     variant('delayed',     default=True, description='Install requirements for dask.delayed (dask.imperative)')
-    variant('yaml',        default=True, description='Ensure support for YAML configuration files')
+    variant('yaml',        default=False, description='Ensure support for YAML configuration files')
 
     conflicts('+distributed', when='@:0.4.0,0.7.6:0.8.1')
     conflicts('+diagnostics', when='@:0.5.0')
-    conflicts('+yaml', when='@:0.17.5')
+    conflicts('+yaml', when='@:0.17.5,2.17.1:')
```

Comment:
The description of the `yaml` variant says "Ensure support for YAML configuration files". The reason it is formulated that way is that we don't really have full control over the feature: it is enabled whenever `py-pyyaml` is available (e.g. as a transitive dependency via `py-distributed`). So, `+yaml` basically says "make sure `py-pyyaml` is installed", which translates to `depends_on('py-pyyaml', when='+yaml')`. I admit that the meaning of `~yaml` is unclear from the description but, to my understanding, the easiest interpretation is "disable the direct dependency on `py-pyyaml`". Since the developers decided to make the dependency on `py-pyyaml` non-optional starting version `2.17.1`, the conflict statements should look like:
```python
conflicts('+yaml', when='@:0.17.5')
conflicts('~yaml', when='@2.17.1:')
```